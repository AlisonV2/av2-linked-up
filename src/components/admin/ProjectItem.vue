<template>
  <div class="card text-center">
    <div class="card-header title">
      {{ project.title }}
    </div>
    <h6 class="card-subtitle mb-2 text-muted" v-if="role === 'client'">{{ project.artistName }}</h6>
    <h6 class="card-subtitle mb-2 text-muted" v-if="role === 'artist'">{{ project.clientName }}</h6>
    <p class="card-text">{{ project.description }}</p>
    <div class="card-footer text-muted">
      {{ createdAt }}
    </div>
  </div>
</template>

<script>
import { format } from 'date-fns';

/**
 * @exports ProjectItem
 * @type {Component}
 * @requires Sentry
 * @vue-data {array} projects
 */
export default {
  name: 'ProjectItem',
  props: ['project', 'time'],
  data() {
    return {
      createdAt: '',
      role: '',
    };
  },
  created() {
    this.formatDate();
    this.getUserRole();
  },
  methods: {
    formatDate() {
      this.createdAt = format(this.time.toDate(), 'dd/MM/yyyy');
    },
    async getUserRole() {
      await this.$store.dispatch('getUserRole').then(() => {
        this.role = this.$store.getters.getUserRole;
      });
    },
  },
};
</script>
