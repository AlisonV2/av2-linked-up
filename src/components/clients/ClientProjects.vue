<template>
  <div class="row"></div>
</template>

<script>
// import * as Sentry from '@sentry/vue';

/**
 * @exports ClientProjects
 * @type {Component}
 * @requires Sentry
 */
export default {
  name: 'ClientProjects',
  data() {
    return {};
  },
  methods: {},
};
</script>

<style></style>
