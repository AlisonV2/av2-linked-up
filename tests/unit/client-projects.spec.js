import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import ClientProjects from '@/components/clients/ClientProjects.vue';
import ProjectItem from '@/components/admin/ProjectItem.vue';

const actions = {
  getClientProjects: jest.fn(),
};

const getters = {
  getClientProjects: jest.fn(),
};

const store = createStore({
  actions,
  getters,
});

const config = {
  data() {
    return {
      projects: [],
    };
  },
  global: {
    plugins: [store],
    components: {
        ProjectItem,
    }
  },
};

/**
 * @module ClientProjectsTest
 */
describe('ClientProjects.vue', () => {
  it('ClientProjects Snapshot', () => {
    const component = shallowMount(ClientProjects, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Check if getClientProjects action is called', async () => {
    const component = shallowMount(ClientProjects, config);
    ClientProjects.created.call(component.vm);
    await component.vm.$nextTick();
    expect(actions.getClientProjects).toHaveBeenCalled;
  });

  it('Check if getClientProjects getter is called', async () => {
    const component = shallowMount(ClientProjects, config);
    ClientProjects.created.call(component.vm);
    await component.vm.$nextTick();
    expect(getters.getClientProjects).toHaveBeenCalled;
  });
});
