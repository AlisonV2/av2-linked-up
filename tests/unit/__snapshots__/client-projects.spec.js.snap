// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientProjects.vue ClientProjects Snapshot 1`] = `
<div
  data-v-app=""
>
  
  <project-status-stub />
  <!--v-if-->
  
</div>
`;
